test/boost/mutation_reader_test.cc

```diff
@@ -3666,6 +3667,72 @@ SEASTAR_THREAD_TEST_CASE(test_clustering_order_merger_in_memory) {
     }
 }
 
+SEASTAR_THREAD_TEST_CASE(test_clustering_order_merger_sstable_set) {
+  sstables::test_env::do_with_async([] (sstables::test_env& env) {
+    storage_service_for_tests ssft;
+    clustering_order_merger_test_generator g;
+
+    auto make_authority = [] (mutation mut, streamed_mutation::forwarding fwd) {
+        return flat_mutation_reader_from_mutations(tests::make_permit(), {std::move(mut)}, fwd);
+    };
+
+    auto make_tested = [s = g._s, pos = dht::ring_position(g.decorated_pk())]
+            (const time_series_sstable_set& sst_set,
+                const std::unordered_set<int64_t>& included_gens, streamed_mutation::forwarding fwd) {
+        auto q = sst_set.make_min_position_reader_queue(
+            [s, pos, fwd] (sstable& sst) {
+                return sst.read_row_flat(s, tests::make_permit(), pos,
+                            s->full_slice(), seastar::default_priority_class(), nullptr, fwd);
+            },
+            [included_gens] (const sstable& sst) { return included_gens.contains(sst.generation()); });
+        return make_clustering_combined_reader(s, tests::make_permit(), fwd, std::move(q));
+    };
+
+    auto seed = tests::random::get_int<uint32_t>();
+    std::cout << "test_clustering_order_merger_sstable_set seed: " << seed << std::endl;
```

Comment:
FYI we already have a configurable seed mechanism. The seed is global for a test-file, it is printed on startup of the test and it is  configurable via `--random-seed`.

Reply:
Tried to use this parameter, but it doesn't change the used seed.
```
$ git grep "random-seed"
test/lib/mutation_source_test.cc:        // In case of errors, reproduce using the --random-seed command line option with the test_runner seed.
test/perf/perf_fast_forward.cc:        ("random-seed", boost::program_options::value<unsigned>(), "Random number generator seed")
test/perf/perf_fast_forward.cc:            auto conf_seed = app.configuration()["random-seed"];
test/perf/perf_fast_forward.cc:            std::cout << "random-seed=" << seed << '\n';
test/perf/perf_simple_query.cc:        ("random-seed", boost::program_options::value<unsigned>(), "Random number generator seed")
test/perf/perf_simple_query.cc:        auto conf_seed = app.configuration()["random-seed"];
test/perf/perf_simple_query.cc:        std::cout << "random-seed=" << seed << '\n';
```

looks like the parameter is ignored by every test except those benchmarks.